docs/functions/util.rst

```diff
@@ -26,6 +26,16 @@ These functions are designed to be used by themes.
 
    The returned string is a fully escaped terminal formatting sequence.
 
+.. function:: _lp_grep_fields(filename, delimiter, keys...) -> var:lp_grep_fields
+
+   Parse the given filename for one key/value pairs of the form "<key><delimiter><value>"
+   (e.g. "this=that") on each line.
+   Sets an associative array containing the parsed values, for each key.
```

Comment:
It no longer returns an associative array, so this doc needs updating.

Make sure to note that doing a for each loop over the results (like `for item in "${lp_grep_fields[@]}"`) is NOT safe, since missing keys will just be skipped, so an array that looks like
```bash
array[0]=foo
array[2]=bar
```
will expand to exactly those two items, and there will be no indication that index 1 is missing. So explicit indexing (and unset variable checking) is needed. As well as using the `_LP_FIRST_INDEX` offset.